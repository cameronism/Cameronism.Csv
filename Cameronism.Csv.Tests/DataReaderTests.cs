using NUnit.Framework;
using System;
using System.Collections.Generic;
using System.Data.SqlClient;
using System.IO;
using System.Linq;
using System.Text;

namespace Cameronism.Csv.Tests
{
    [TestFixture]
    public class DataReaderTests
    {
        private static void ExecuteReader(string sql)
        {
            using (var conn = new SqlConnection(@"Server=(localdb)\MSSQLLocalDB;Integrated Security=true"))
            {
                conn.Open();
                using (var cmd = conn.CreateCommand())
                {
                    cmd.CommandText = sql;
                    using (var reader = cmd.ExecuteReader())
                    {
                        var sw = new StringWriter { NewLine = "\r\n" };
                        Serializer.Serialize(sw, reader);
                        Approvals.VerifyCsv(sw);
                    }
                }
            }
        }

        [Test]
        public void OneString()
        {
            ExecuteReader("select 'bar' as foo");
        }

        [Test]
        public void TwoString()
        {
            ExecuteReader("select 'ba\nr\"' as foo, 1 as one, null as n");
        }

        [Test]
        public void SomeBytes()
        {
            ExecuteReader("select 0xFF0872FDDA3FC1EFDA9706B2B3FBAC7BF6DAB1EEB8 as bytes");
        }
    }
}
