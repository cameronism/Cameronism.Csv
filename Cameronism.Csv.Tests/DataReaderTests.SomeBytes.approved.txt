bytes
/why/do/we/always/use/base64
